crates/ruff_linter/src/linter.rs

```diff
@@ -260,6 +260,29 @@ pub fn check_path(
         }
     }
 
+    // Update fix applicability to account for overrides
+    if !settings.extend_safe_fixes.is_empty() || !settings.extend_unsafe_fixes.is_empty() {
+        for diagnostic in &mut diagnostics {
+            if let Some(fix) = diagnostic.fix.take() {
+                // Check unsafe before safe so if someone puts a rule in both we are conservative
+                if settings
+                    .extend_unsafe_fixes
+                    .contains(diagnostic.kind.rule())
+                    && fix.applicability().is_safe()
```

Comment:
I think it's worth inverting the order of the `&&` here, since the second condition is much cheaper.

Reply:
Hm I thought I was being careful to do the cheaper option first but it sure doesn't look like it!